docs/book/v2/dispatching-middleware.md

```diff
@@ -0,0 +1,272 @@
+
+## Closures
+
+A `Closure` can also be passed directly. This should also be used only for development.
+
+```php
+use Laminas\Mvc\Middleware\PipeSpec;
+use Laminas\Router\Http\Literal;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+return [
+    'router' => [
+        'routes' => [
+            'closure' => [
+                'type'    => Literal::class,
+                'options' => [
+                    'route'    => '/closure',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => static function (
+                            ServerRequestInterface $request,
+                            RequestHandlerInterface $handler
+                        ): ResponseInterface {
+                            return $handler->handle($request);
+                        },
+                    ],
+                ],
+            ],
+        ],
+    ],
+];
+```
+
+## Mixed `PipeSpec` Arguments
+
+In this example multiple different possible parameters are passed to `PipeSpec`:
+
+```php
+use Application\Handler\AlbumDetailHandler;
+use Laminas\Mvc\Middleware\PipeSpec;
+use Laminas\Router\Http\Literal;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+return [
+    'router' => [
+        'routes' => [
+            'instance-pipespec-dev-env' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/instance-pipespec-dev-env',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => new PipeSpec(
+
+                            // id/alias of service in container
+                            'container_id_for_album_middleware',
+
+                            // anonymous class implementing a MiddlewareInterface
+                            new class implements MiddlewareInterface {
+                                public function process(
+                                    ServerRequestInterface $request,
+                                    RequestHandlerInterface $handler
+                                ): ResponseInterface {
+                                    return $handler->handle($request);
+                                }
+                            },
+
+                            // Closure
+                            static function (
+                                ServerRequestInterface $request,
+                                RequestHandlerInterface $handler
+                            ): ResponseInterface {
+                                return $handler->handle($request);
+                            },
+
+                            // class-string of an Handler which will be resolved by the container
```

Comment:
This might imply that such usage will use auto-invokable feature or some kind of abstract factory. No such feature is provided however.
This might be a more clear way to describe the usage: 
```suggestion
                            // class-string of a handler as a convenient way to specify id/alias which must be resolved by the container
```